src/analysis/normed_space/inner_product.lean

```diff
@@ -2304,6 +2332,38 @@ variables (K)
 /-- Reflection is involutive. -/
 lemma reflection_involutive : function.involutive (reflection K) := reflection_reflection K
 
+variables {K}
+
+/-- A point is its own reflection if and only if it is in the subspace. -/
+lemma reflection_eq_self_iff (x : E) : reflection K x = x ↔ x ∈ K :=
+begin
+  rw [←orthogonal_projection_eq_self_iff, reflection_apply, sub_eq_iff_eq_add', ← two_smul 𝕜,
+    ← two_smul' 𝕜],
+  refine (smul_right_injective E _).eq_iff,
+  exact two_ne_zero
```

Comment:
```suggestion
  exact (smul_right_injective E is_R_or_C.two_ne_zero).eq_iff,
```
Maybe this is worse, because we have to write out `is_R_or_C`, or maybe it's better for the same reason. :-) Not a big deal either way.